### DISCUSSION TITLE
High Availability

### DISCUSSION THREAD

**Discussion by ritik872000** [concourse/concourse#8269] [Help & Support]
Hi @taylorsilva 
I have set up 2 servers. 
when one server goes down I want the workers which were running on the server to shift on the server which is running.

any kind of help would be appreciated.

Many thanks!
Ritik

**Comment by taylorsilva** [comment]
That's outside the scope of what Concourse would do. However you're deploying your workers you'll need that thing to see "two workers disappeared. Better make some new ones"

A second way I could interpret your question is you mean you want everything that was on the workers that disappeared to start running on other workers that are still running. There is a feature you can enable, `--enable-rerun-when-worker-disappears`, that tries to address this (I think).

**Reply by ritik872000** [reply]
Hi @taylorsilva, thanks for the reply, 

so we've already enabled this feature which you've mentioned.

currently, we are trying to setup peer address so, like if a job is assigned to a worker and that worker disconnects, it should handover that job to another worker using the peer address. 

also, how can I configure peer adress? I am unable to find any concrete documentation regarding the peer address.

Thanks in advance:)

**Reply by taylorsilva** [reply]
You're like the third or fourth person that has asked for docs for p2p in the last few weeks so I finally sat down and wrote them out. Enjoy!

https://concourse-ci.org/concourse-worker.html#configuring-peer-to-peer-volume-streaming

**Reply by ritik872000** [reply]
Hi, 

Thanks for providing the documentation!

I was actually asking related to high availability setup with 2 web-server nodes. The problem we're facing is related to this variable "CONCOURSE_PEER_ADDRESS".

is there any documentation where two web server node setup instructions are given? and what should be the CONCOURSE_PEER_ADDRESS to be configures in both web-server nodes.

Many thanks!

**Reply by taylorsilva** [reply]
ah gotcha! That's covered in the docs here: https://concourse-ci.org/concourse-web.html#scaling

Though I can see how they're a bit unclear. I'll clarify them with an example.

let's say you have two web nodes with the following LAN IP's:

```
web1 - 10.0.1.3
web2 - 10.0.1.8
```

You would configure the web nodes with the following peer address:

```
web1
CONCOURSE_PEER_ADDRESS=10.0.1.3

web2
CONCOURSE_PEER_ADDRESS=10.0.1.8
```

Hope that helps!

**Reply by deepak2717** [reply]
Hi Taylorsilva,
Thanks for the response.

I tried with your suggestion. The problem is when web1 goes down the the registered workers to web1 do not shift to web2 node. The jobs starts failing  with the hanging workers of web1.

Any suggestions will be a great help.

Many Thanks!

**Reply by deepak2717** [reply]
We see this error : Oct 21 05:32:31 ip-11-221-254-200 concourse-server[20372]: {"timestamp":"2022-10-21T05:32:31.752159326Z","level":"debug","source":"tsa","message":"tsa.connection.channel.command.end","data":{"command":"report-volumes","remote":"11.221.254.56:22660","session":"25.1.1"}}
Oct 21 05:32:31 ip-11-221-254-200 concourse-server[20372]: {"timestamp":"2022-10-21T05:32:31.752178391Z","level":"error","source":"tsa","message":"tsa.connection.channel.exited-with-error","data":{"error":"bad-response (500): HTTP/1.1 500 Internal Server Error\r\nContent-Length: 0\r\nCache-Control: no-store, private\r\nContent-Security-Policy: frame-ancestors 'none'\r\nContent-Type: application/json\r\nDate: Fri, 21 Oct 2022 05:32:31 GMT\r\nVary: Accept-Encoding\r\nX-Concourse-Version: 7.7.1\r\nX-Content-Type-Options: nosniff\r\nX-Download-Options: noopen\r\nX-Frame-Options: deny\r\nX-Xss-Protection: 1; mode=block\r\n\r\n","remote":"11.221.254.56:22660","session":"25.1"}}
